src/raftstore/store/worker/apply.rs

```diff
@@ -515,15 +463,70 @@ pub struct ApplyDelegate {
     pending_cmds: PendingCmdQueue,
     metrics: ApplyMetrics,
     last_merge_version: u64,
+
+    scheduler: Scheduler<Task>,
+    notifier: Sender<TaskRes>,
+    host: Arc<CoprocessorHost>,
+    importer: Arc<SSTImporter>,
+    use_delete_range: bool,
+
+    stopping: bool,
 }
 
+impl Debug for ApplyDelegate {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        f.debug_struct("ApplyDelegate")
+            .field("tag", &self.tag)
+            .field("id", &self.id)
+            .finish()
+    }
+}
+
+type RaftEntryFutureItem = (ApplyDelegate, ApplyContext, Option<ExecResult>);
```

Comment:
can you use field structure instead of the tuple?

Reply:
I suggest using field structure for more clear instead of tuple all the times. Unless your tuple has little items (I even think must <= 2) and you always destruct the tuple for outside use.   

Btw, for a tuple, what tuple.0, tuple.1 really mean?

Reply:
Because these types are used in future, things have some differences, let's see the code where these types are used:
```
delegate.handle_raft_entry(apply_ctx, entry)
    .and_then(move |(delegate, apply_ctx, exec_result)| {
        ...
})
```
If we use a struct to wrap it, the code is like this:
```
delegate.handle_raft_entry(apply_ctx, entry)
    .and_then(move |res| {
        res.delegate ...
        res.apply_ctx ...
        res.exec_result ...
})
```
So you can see the difference.